File: Interface/src/app/interfaces/event.ts
import { Schedule } from "./schedule";

export class DeviceEvent {
    
    deviceID: string;
    level: string;
    _id: string;
    message: string;
    ownerID: string;
    title: string;
    createTime: number;
    details: string[] = [];

    constructor(id: string, deviceID: string, level: string, message: string, ownerID: string, title: string, createTime: number, details: string[]) {
        this._id = id;
        this.deviceID = deviceID;
        this.level = level;
        this.message = message;
        this.ownerID = ownerID;
        this.title = title;
        this.createTime = createTime;
        this.details = details;
    }

}

File: System/Base/Entities/Event.py
from datetime import date

class Event:
    
    ownerID: str
    deviceID: str
    title = ''
    message = ''
    level = ''

    def __init__(self, ownerID: str, deviceID: str, title = None, message = None, level = None):
        self.deviceID = deviceID
        self.ownerID = ownerID
        self.title = title
        self.message = message
        self.level = level

    def get_deviceID(self):
        return self.deviceID

    def set_deviceID(self, value):
        self.deviceID = value

    def get_ownerID(self):
        return self.ownerID

    def set_ownerID(self, value):
        self.ownerID = value

    def get_title(self):
        return self.title

    def set_title(self, value):
        self.title = value

    def get_message(self):
        return self.message

    def set_message(self, value):
        self.message = value

    def get_level(self):
        return self.level

    def set_level(self, value):
        self.level = value


    